### ISSUE TITLE
kloudless.files.contents() returns corrupted stream

### ISSUE THREAD

**Issue by james-wagner11** [Kloudless/kloudless-node#10]
I'm using `kloudless.files.contents` to get a download stream and pipe it to S3, but the output files in S3 are corrupt. The same issue can be seen when piping the stream to a local file. For example, when I download a jpg, the output file is corrupt:
```javascript
kloudless.files.contents({
  account_id: [account_id],
  file_id: [file_id]
}, (err, stream) => {
  if(err){
    console.log(err);
  } else {
    var output = require('fs').createWriteStream('./output.jpg');
    stream.pipe(output);
  }
});
```

I can fix the issue by removing the line `res.setEncoding('binary');` in `lib/resources.js` (https://github.com/Kloudless/kloudless-node/blob/master/lib/resources.js#L44) and the files download properly, but I'm not sure what other consequences there may be to that. Do you think this fix is appropriate?

**Comment by hong19** [comment]
@james-wagner11 You are right, the bug is caused by `res.setEncoding('binary');` in `lib/resources.js` (https://github.com/Kloudless/kloudless-node/blob/master/lib/resources.js#L44)
It use the method of library [request](https://github.com/request/request#streaming). The explication of `setEncoding` is in below section.

---
- `encoding` - encoding to be used on `setEncoding` of response data. If `null`, the `body` is returned as a `Buffer`. Anything else **(including the default value of `undefined`)** will be passed as the [encoding](http://nodejs.org/api/buffer.html#buffer_buffer) parameter to `toString()` (meaning this is effectively `utf8` by default). (**Note:** if you expect binary data, you should set `encoding: null`.)
---

`res.setEncoding('binary');` will cause the response data as `string` not `Buffer`.
Depend on the docs, it should be `res.setEncoding(null);`.  
But in my testcases, it doesn't read the stream as `Buffer` when `res.setEncoding(null);`, but removing `res.setEncoding();` will get response data `Buffer` correctly.

I will remove that line and write a testcase for that.  Thank you for helping find the bug.

**Comment by hong19** [comment]
@james-wagner11 Hello, the this bug has been fixed, please check bd84e5f9bcd007deae1e8d2bbd39e4766f046576

**Comment by vinodc** [comment]
1.0.5 published.

Thanks for the bug report and the resolution!
